gaps/gap-draft_experimental_features/gap-draft_experimental_features.md

```diff
@@ -0,0 +1,263 @@
+- Development version (PoC)
+- MVP release
+- Iterating over released feature
+- Upgrade to stable version
+
+Before upgrading feature, following things should be considered or taken care of:
+- Backward compatibility with previous versions of protocol (and potential interactions/collisions)
+- Forward compatibility and reasonable level of extensibility
+- Security
+- Stability and quality
+- Generality level: **Stable Features** shouldn't contain implementation solutions that close
+  protocol usage for entities outside of Golem Factory
+
+Notice that _upgrade process_ from **Experimental Feature** to **Stable Feature** is not backward
+compatible itself, although developers can choose to keep compatibility if they see reasons for it.
+
+### Experimental features separation
+
+**Experimental features** should be separated on protocol level by adding specific prefix
+and creating special _experimental_ namespace this way. Everything within this namespace
+can be changed and any time and doesn't have stable semantic. Moreover, everyone is allowed
+to use names from experimental namespace.
+
+**Stable Features** protocols aren't allowed to use reserved prefixes, otherwise they will be
+treated as experimental.
+
+In following paragraphs experimental namespacing for different kind of APIs will be introduced.
+Moreover, a few guidelines will be proposed to avoid naming collisions.
+
+Keep in mind, that since everything within _experimental_ namespace is undefined, no one
+is forced to keep the guidelines, if they would hinder ability to fast development.
+
+#### Negotiation protocols
+
+Assuming we have `golem.com.payment`, let's introduce experimental `payment-platform` namespace:
+
+`"golem.com.payment.!exp.payment-platform" : "erc20-rinkeby-tglm"`
+
+(TODO: Check if we can use `!` sign. May cause conflict in constraint language,
+json or yaml Offer/Demand representation)
+
+We have multiple options of correct experimental namespaces:
+- `"golem.com.payment.!exp.payment-platform"`
+- `"golem.com.!exp.payment.payment-platform"`
+- `"golem.!exp.com.payment.payment-platform"`
+- `"!exp.golem.com.payment.payment-platform"`
+
+Example of incorrect namespacing:
+
+`"golem.com.payment.payment-platform.!exp" : "erc20-rinkeby-tglm"`
+
+`payment-platform` property didn't exist before, so we are not allowed to introduce it.
+
+#### GSB protocols
+
+GSB allows to make RPC calls to other yagna daemons. Messages are sent to specific address
+like for example:
+`/public/market/protocol/mk1/discovery/offers/GetOffers`
+
+Examples of introducing new GSB message `ListOffers`:
+- `/public/!exp/market/protocol/mk1/discovery/offers/ListOffers`
+- `/public/market/!exp/protocol/mk1/discovery/offers/ListOffers`
+- `/public/market/protocol/!exp/mk1/discovery/offers/ListOffers`
+- `/public/market/protocol/mk1/!exp/discovery/offers/ListOffers`
+- `/public/market/protocol/mk1/discovery/!exp/offers/ListOffers`
+- `/public/market/protocol/mk1/discovery/offers/!exp/ListOffers`
```

Comment:
Similarly to properties, I think enforcing experimental namespace into `/public/!exp/...` means easier grouping.

Reply:
I added to good practices, but I'm not sure if I wouldn't prefer `/public/market/!exp/...`